src/apps/tftp/tftp.c

```diff
@@ -454,10 +454,12 @@ tftp_init_common(u8_t mode, const struct tftp_context *ctx)
     return ERR_MEM;
   }
 
-  ret = udp_bind(pcb, IP_ANY_TYPE, TFTP_PORT);
-  if (ret != ERR_OK) {
-    udp_remove(pcb);
-    return ret;
+  if (mode == LWIP_TFTP_MODE_SERVER) {
```

Comment:
since we have LWIP_TFTP_MODE_CLIENTSERVER, maybe check if the server bit is set?

Reply:
Absolutely. Patch updated.